File: Sources/MauItKit/Models/SignedFormatting.swift
import Foundation

/// The app never uses a bare ASCII hyphen for a negative number — every
/// signed figure (burn credit, over-budget delta, weight change, average
/// balance) gets an explicit `+` or a typographic minus.
public enum SignedFormatting {

    public static func integer(_ value: Int) -> String {
        value < 0 ? "\u{2212}\(abs(value))" : "+\(value)"
    }

    public static func decimal(_ value: Double, fractionDigits: Int = 1) -> String {
        let magnitude = String(format: "%.\(fractionDigits)f", abs(value))
        return value < 0 ? "\u{2212}\(magnitude)" : "+\(magnitude)"
    }
}

File: packages/shared/src/signedFormatting.ts
/**
 * The app never uses a bare ASCII hyphen for a negative number — every
 * signed figure (burn credit, over-budget delta, weight change, average
 * balance) gets an explicit `+` or a typographic minus.
 */

export function formatSignedInteger(value: number): string {
  return value < 0 ? `−${Math.abs(value)}` : `+${value}`;
}

export function formatSignedDecimal(value: number, fractionDigits = 1): string {
  const magnitude = Math.abs(value).toFixed(fractionDigits);
  return value < 0 ? `−${magnitude}` : `+${magnitude}`;
}
